pygmt/datasets/tile_map.py

```diff
@@ -176,8 +194,12 @@ def load_tile_map(
         dims=("band", "y", "x"),
     )
 
-    # If rioxarray is installed, set the coordinate reference system
+    # If rioxarray is installed, set the coordinate reference system.
     if hasattr(dataarray, "rio"):
-        dataarray = dataarray.rio.write_crs(input_crs="EPSG:3857")
+        dataarray = dataarray.rio.write_crs(input_crs=_default_crs)
+
+    # Reproject raster image from EPSG:3857 to specified CRS, using rioxarray.
+    if crs != _default_crs:
+        dataarray = dataarray.rio.reproject(dst_crs=crs)
```

Comment:
I realize there's an `if crs != _default_crs and not _HAS_RIOXARRAY:` statement at the top, so it's ok to keep this on the outer indent, but I still think it's safer to nest this under the `if hasattr(dataarray, "rio"):` statement.

```suggestion
        dataarray = dataarray.rio.write_crs(input_crs=_default_crs)

        # Reproject raster image from EPSG:3857 to specified CRS, using rioxarray.
        if crs != _default_crs:
            dataarray = dataarray.rio.reproject(dst_crs=crs)
```